### ISSUE TITLE
epiweeks packages

### ISSUE THREAD

**Issue by aspina7** [appliedepi/epiRhandbook_eng#34]
just flagging for @nsbatra and @henryls1 so we can unify all the bits with dates.
**tldr:** stick with lubridate not aweek 

mainly that there are multiple ways of doing epiweeks and that {[aweek](https://github.com/reconhub/aweek)} might not be our best option anymore - and I think based on [discussions ](https://github.com/reconhub/aweek/issues/34) with zhian he is probably in favour of retiring it going forward. 

The main advantage of {aweek} over others was that it allowed for defining weeks with differing start days (i.e. not ISO or CDC standard monday or monday) - which is useful in several countries. The disadvantage is that doing calculations is a bit more painful (but possible - see tim's post later down). 

It turns out that with {[lubridate](https://lubridate.tidyverse.org/reference/day.html)} there is an option to set the start day and that has been integrated in the {[tsibble](https://tsibble.tidyverts.org/)} package - see this [discussion](https://github.com/tidyverts/tsibble/issues/205). 

The other side of the coin is that {[incidence](https://github.com/reconhub/incidence)} and the new {[incidence2](https://github.com/reconhub/incidence2)} for making epicurves is built off the back of {aweek} - see [tim's reasoning](https://github.com/reconhub/incidence2/issues/13).  

So i think we are probably best off sticking to {lubridate} as henry has done (not using {aweek} for defining weeks) - and then in the time series bit I will use {tsibble} which plays nice with {lubridate}. 
{incidence} inputs is the date vars anyway so do not need to define weeks in advance so I think we are okay there. 

sorry for the rant - thoughts?

**Comment by henryls1** [comment]
Hi Alex, I've actually never used {aweek} before, and always just go with lubridate. I find that I can do everything I ever have needed with isoweek() and epiweek() and floor_date(), but as you say, this won't work for countries that use days that aren't sun/mon to start the week. 

I'd generally agree that {lubridate} is sufficient, but we might want to include an example for cases with non-ISO/EPI start days, using floor_date() or something.

**Comment by aspina7** [comment]
hey henry - agee, think it should be possible with something like `options(lubridate.week.start = 5)` but just fiddling now ...

**Comment by henryls1** [comment]
I played around a bit as well, and while options(lubridate.week.start = 5)
works for floor_date() and wday(), it doesn't seem to affect week()  (I
supposed intentionally). So it seems as if there is no isoweek()/epiweek()
equivalent in lubridate for non-standard week starts. I don't know if these
functions are actually essential though honestly

**Comment by henryls1** [comment]
For most purposes i find a variable giving the week start date much more
valuable than a numeric value giving the week of the year.

**Comment by aspina7** [comment]
yeah you are right - there is a filthy workaround in lubridate but its not pretty. 
In general I would agree with you, and just the numeric week as produced by lubridate is really not helpful. 
{aweek} and {tsibble} produce more useful year and week outputs.... 

The two instances where I think using weeks over first date of week is useful is: 
1) for automating reports - is easier to define the current week of interest at the top of an rmd and update each week (rather than looking up the appropriate date each week) - WHO and MSF both still producing reports by week. 
2) Plays nice with time series (doing lags etc less dense...) and forecasting - and axis labels feel less dense with year and week than with full dates (might just be me tho..) 

```
aweek::date2week("2020-01-01", week_start = "Friday")
#> <aweek start: Freitag>
#> [1] "2019-W52-6"

tsibble::yearweek("2020-01-01", week_start = 5)
#> <yearweek[1]>
#> [1] "2019 W52"
#> # Week starts on: Friday

lubridate::isoweek(
     lubridate::floor_date(
          lubridate::ymd("2020-01-01"),
          unit = "week", 
          week_start = getOption("lubridate.week.start", 5)))
#> [1] 52
Created on 2021-01-10 by the reprex package (v0.3.0)
```

**Comment by nsbatra** [comment]
Thanks all for the robust discussion @aspina7  @henryls1 . Adding @isaac-florence because I saw he used lubridate in the plotly page.

I'm ok moving away from aweek if we are feeling that direction, especially if pckg development/support is on the downslope. We can always mention aweek as an option, explain pros/cons, but de-emphasize.  E.g. I wrote content on this in the Epicurves page (see tab on ggplot, sub-tab Modifications, sub-tab Aggregating from Linelist). There I offered the user aweek and lubridate methods, but I can switch the emphasis and primary examples to lubridate.

A few thoughts however:  
- I do really appreciate how easily aweek achieves YYYY-Www.  
- I like aweek's simple *factor functionality* (just `factor=T`) so that when summarizing counts *all* weeks are included - even those with no cases. To me this is a pretty big deal. Equivalent code in lubridate seems to need addition of complete() afterwards to achieve the same, which is cumbersome ... *How important is this aspect to you all?*  
```
# Make dataset of weekly case counts
weekly_counts <- linelist %>% 
  mutate(week= lubridate::floor_date(date_onset, unit = "week")) %>%   # new column, week of onset
  group_by(week) %>%                                                                              # group data by new month col, keep all weeks
  summarise(n_weekly_cases = n()) %>%                                                   # new summary column
  complete(month = seq.Date(min(week),                                                  # fill-in all weeks with no cases reported
                            max(week),
                            by="week"))
```
- On the other hand, usinglubridate floor_date() means we'd use the same function for aggregating to weeks as for months. The consistency is nice.

**Comment by isaac-florence** [comment]
I'd personally lean towards lubridate because - as mentioned above - one can use `lubridate::floor_date(Sys.Date(), "week", week_start = 3)` (the use of getOption is superfluous it seems) for countries not using mon or sun as week starts, you can work with more than just weeks but also months, quarters, etc, as well as the fact we're nicely tidy verse centric in this book and all tidyverse vignettes and SO help will point to lubridate. 

Although I agree using tidy::complete is a pain, I think it's worth knowing about in other contexts too and so is worth teaching anyway.

**Comment by nsbatra** [comment]
Alright I'm convinced. I'll switch my Epicurves aweek code to lubridate, unless there are still objections. But let's still give mentions of aweek in most places noting its utility for producing YYYY-Www. These notes can link to a somewhat thorough explanation of aweek in the Aggregating/Grouping page (@sarahollis)
 
Good point @isaac-florence - I also simply used the `week_start = ` argument within floor_date() in my examples of lubridate.

**Comment by aspina7** [comment]
Ah nice thanks @isaac-florence 
Ye agree @nsbatra about yyyywww being nice. I'd favour tsibble for this over aweek, as also uses lubridate.... But not stuck to this...

**Comment by nsbatra** [comment]
@aspina7  ah right!! sorry missed that. Yes especially if tsibble is being used elsewhere for time series, let's minimize the number of packages. So we can cut out aweek. 
Makes me a bit sad, but this world changes fast!

**Comment by nsbatra** [comment]
Following up on this. I just converted the epicurves page to be lubridate-centric with an alternative dish of aweek
